modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxNdfTradeCsvPlugin.java

```diff
@@ -0,0 +1,159 @@
+  public Set<String> tradeTypeNames() {
+    return ImmutableSet.of("FXNDF", "FX NDF", "NDF");
+  }
+
+  @Override
+  public Optional<Trade> parseTrade(
+      Class<?> requiredJavaType,
+      CsvRow baseRow,
+      List<CsvRow> additionalRows,
+      TradeInfo info,
+      TradeCsvInfoResolver resolver) {
+
+    if (requiredJavaType.isAssignableFrom(FxNdfTrade.class)) {
+      return Optional.of(resolver.parseFxNdfTrade(baseRow, info));
+    }
+    return Optional.empty();
+  }
+
+  @Override
+  public String getName() {
+    return "FxNdf";
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Parses from the CSV row.
+   *
+   * @param row the CSV row
+   * @param info the trade info
+   * @param resolver the resolver used to parse additional information
+   * @return the parsed trade
+   */
+  static FxNdfTrade parse(CsvRow row, TradeInfo info, TradeCsvInfoResolver resolver) {
+    FxNdfTrade trade = parseRow(row, info);
+    return resolver.completeTrade(row, trade);
+  }
+
+  //-------------------------------------------------------------------------
+  // parses the row to a trade
+  private static FxNdfTrade parseRow(CsvRow row, TradeInfo info) {
+    CurrencyAmount settlementCurrencyNotional = CsvLoaderUtils.parseCurrencyAmountWithDirection(
+        row, SETTLEMENT_CURRENCY_FIELD, SETTLEMENT_CURRENCY_NOTIONAL_FIELD, SETTLEMENT_CURRENCY_DIRECTION_FIELD);
+    LocalDate paymentDate = row.getField(PAYMENT_DATE_FIELD, LoaderUtils::parseDate);
+    Currency settlementCurrency = row.getField(SETTLEMENT_CURRENCY_FIELD, LoaderUtils::parseCurrency);
+    Currency nonDeliverableCurrency = row.getField(NON_DELIVERABLE_CURRENCY_FIELD, LoaderUtils::parseCurrency);
+    CurrencyPair currencyPair = CurrencyPair.of(settlementCurrency, nonDeliverableCurrency).toConventional();
+    FxRate agreedFxRate = FxRate.of(currencyPair, row.getField(FX_RATE_FIELD, LoaderUtils::parseDouble));
+    FxIndex.extendedEnum();
+    FxIndex index = parseFxIndex(currencyPair)
+        .orElseThrow(() -> new IllegalArgumentException(Messages.format(
+        "No FX Index found for currency pair {}. Known FX Index required to construct NDF trade",
+        currencyPair.toString())));
+    FxNdf fxNdf = FxNdf.builder()
+        .settlementCurrencyNotional(settlementCurrencyNotional)
+        .agreedFxRate(agreedFxRate)
+        .index(index)
+        .paymentDate(paymentDate)
+        .build();
+    return FxNdfTrade.of(info, fxNdf);
+  }
+
+  // parses the FX Index from a currency pair
+  private static Optional<FxIndex> parseFxIndex(CurrencyPair currencyPair) {
+    ImmutableMap<CurrencyPair, FxIndex> ccyFxIndexMap = FxIndex.extendedEnum().lookupAll().values().stream()
+        .collect(toImmutableMap(
+            fxIndex -> fxIndex.getCurrencyPair().toConventional(),
+            fxIndex -> fxIndex,
+            (fxIndex1, fxIndex2) -> fxIndex1)); //arbitrarily chose when we have multiple indices for the same pair
```

Comment:
The FX index name should be in the csv format. Potentially it should be in the convention column. If this is done, there is probably no need for the separate settlement and ND currency columns.

There is also a good case for re-using the Buy/Sell column, instead of adding the new Pay/Receive one

Reply:
The way we always end up parsing these trades is to take two currencies and work out the index ourselves; typically the  specific index is not something that clients will have available in their trade capture systems (it will be part of trade confirms). 

Whilst technically Strata is correct in terms of modelling the trade it is not really practical to force all users to map to the index names that we recognise.

We could put this in a private repo instead?

Reply:
What is needed is a way to allow a currency pair to be entered in the convention column and mapped to a suitable FX index (ideally just a generic one that doesn't provide any more detail). When writing it out, and for round tripping, the full index name must be used.

Reply:
Perhaps we try to search the `FxIndexData.csv` and if not present, we create using a `ImmutableFxRate` with suitable defaults?

Reply:
Yes, `FxIndexCsvLookup` can probably be extended to support creating indices on the fly for plain currency-pairs